src/lib/ndpi_main.c

```diff
@@ -100,10 +100,16 @@ void *ndpi_calloc(unsigned long count, size_t size) {
 /* ****************************************** */
 
 void ndpi_free(void *ptr) {
```

Comment:
If your patch is necessary then it means that we're freeing NULL pointers. Can you explain here this happens in the code? Better to fix the root cause.

Reply:
if the ptr is NULL,it's not will be free;In the file ndpi_classify.c ,function ndpi_classify, use the ndpi_free  will have the risk free NULL pointer
`if(!merged_lens || !merged_times) {
    ndpi_free(merged_lens);
    ndpi_free(merged_times);
    return(score);
  }`